### ISSUE TITLE
error with gzipped fastq file and create working dir if it doesn't exist

### ISSUE THREAD

**Issue by wdecoster** [eldariont/svim#1]
Hi,

This looks like an interesting tool so I'm eager to give it a try.

Using a gzipped fastq file I get the following error:

```
Traceback (most recent call last):
  File "svim/SVIM_COLLECT.py", line 383, in <module>
    sys.exit(main())
  File "svim/SVIM_COLLECT.py", line 353, in main
    full_reads_path = create_full_file(options.working_dir, options.reads, reads_type)
  File "svim/SVIM_COLLECT.py", line 200, in create_full_file
    if line.startswith('@'):
TypeError: startswith first arg must be bytes or a tuple of bytes, not str
```

I could solve this by changing, in line 178 of SVIM_COLLECT.py the mode of `gzip.open()` from 'rb' to 'rt'.

Additionally, would it be possible that the script creates the working dir if it doesn't exist? Earlier I had the following error:

```Traceback (most recent call last):
  File "svim/SVIM_COLLECT.py", line 383, in <module>
    sys.exit(main())
  File "svim/SVIM_COLLECT.py", line 321, in main
    fileHandler = logging.FileHandler("{0}/SVIM-COLLECT_{1}.log".format(options.working_dir, strftime("%y%m%d_%H%M%S", localtime())), mode="w")
  File "/home/wdecoster/anaconda3/envs/svim/lib/python3.6/logging/__init__.py", line 1030, in __init__
    StreamHandler.__init__(self, self._open())
  File "/home/wdecoster/anaconda3/envs/svim/lib/python3.6/logging/__init__.py", line 1059, in _open
    return open(self.baseFilename, self.mode, encoding=self.encoding)
FileNotFoundError: [Errno 2] No such file or directory: '/home/wdecoster/temp/test_svim/test/SVIM-COLLECT_180620_125723.log'
```
which was solved by creating the 'test' directory.

Cheers,
Wouter

**Comment by eldariont** [comment]
Hi Wouter,

thanks for these very good suggestions! :)

I fixed the gzip bug in fabff53ecc6c7fcd6b54641a5fa8570bbc7c48c8 and added automatic creation of the working dir in b58b319f19d3f1b27aa5290a26939c32ee247060.

Cheers,
David

**Comment by wdecoster** [comment]
Thanks for the quick fix!
